.openshift-ci/jobs/integration-tests/env.sh

```diff
@@ -33,3 +38,18 @@ shopt -s nullglob
 for cred in /tmp/secret/**/[A-Z]*; do
     export "$(basename "$cred")"="$(cat "$cred")"
 done
+
+# only run all integration tests on master, or when the all-integration-tests
+# label is added. This is checked in this common env script because it allows
+# us to skip pre- and post- steps as well (which source this file)
+if ! pr_has_label "all-integration-tests" && is_in_PR_context; then
+    if [[ ! "$IMAGE_FAMILY" =~ (rhel-(7|8)|ubuntu-) ]]; then
+        echo "Not running integration tests for ${VM_CONFIG}"
+        exit 0
+    fi
+fi
+
+if pr_has_label "skip-integration-tests"; then
+    echo "Skipping integration tests for ${VM_CONFIG}"
+    exit 0
+fi
```

Comment:
[ultra-nit] It might make sense to make this the very first check, since it will skip all integration tests disregarding any other conditions.